aten/src/ATen/native/sparse/SparseUtils.h

```diff
@@ -0,0 +1,127 @@
+#include <ATen/ATen.h>
+#include <ATen/SparseTensorImpl.h>
+
+namespace at { namespace native {
+
+// Just for documentary purposes
+using SparseTensor = Tensor;
+using LongTensor = Tensor;
+using IntTensor = Tensor;
+using SparseType = Type;
+
+namespace {
+
+// This is an internal utility function for getting at the SparseTensorImpl,
+// so that we can write sparse tensor specific accessors for special fields
+// in SparseTensor.  You should only use this for writing low level
+// setters/getters for SparseTensorImpl fields; otherwise, you should use
+// the low level setters/getters that were implemented using this.
+//
+// This may be called repeatedly, so make sure it's pretty cheap.
+SparseTensorImpl* _get_sparse_impl(const SparseTensor& self) {
+  if (!self.is_sparse()) AT_ERROR("_internal_get_SparseTensorImpl: not a sparse tensor");
+  return static_cast<SparseTensorImpl*>(self.unsafeGetTensorImpl());
+}
+
+// Port of the old THCSTensor_(checkGPU), but it doesn't really belong here
+// because it is more general
+// NB: I dropped kernelP2PEnabled support
+// NB: This does NOT assume that the tensors are GPU tensors; if they're
+// all CPU that's OK too
+inline bool _check_device(ArrayRef<Tensor> ts) {
+  if (ts.empty()) {
+    return true;
+  }
+  const Tensor& ref_t = ts.front();
+  int64_t curDevice = -1;
+  if (ref_t.is_cuda()) {
+    curDevice = current_device();
+  }
+  for (const Tensor& t : ts) {
+    if (t.get_device() != curDevice) return false;
+  }
+  return true;
+}
+
+inline void _raw_resize_sparse(const SparseTensor& self, int64_t sparseDims, int64_t denseDims, ArrayRef<int64_t> size) {
+  _get_sparse_impl(self)->raw_resize_(sparseDims, denseDims, size);
+}
+
+// Takes indices and values and directly puts them into the sparse tensor, no
+// copy.  This used to be called THSTensor_(_move)
+inline void _alias_into_sparse(const SparseTensor& self, const LongTensor& indices, const Tensor& values) {
+  _get_sparse_impl(self)->set_indices_and_values(indices, values);
+}
+
+// Take indices and values and makes a (data) copy of them to put into the sparse
+// indices/values.  This used to be called THSTensor_(_set)
+inline void _copy_into_sparse(const SparseTensor& self, const LongTensor& indices, const Tensor& values) {
+  _alias_into_sparse(self, indices.clone(), values.clone());
+}
+
+// Does NOT make copies of indices/values
+inline SparseTensor _new_with_dims_and_tensor_sparse(
+    const SparseType& dtype,
+    int64_t sparseDims,
```

Comment:
nit: we don't use camel case elsewhere.

Reply:
So, I used to snake-ify the variable names as I converted them, but eventually it just got really annoying (because I had to snake-ify the use sites and it was just easier to retain the original variable names), so I stopped snake-ifying them. TBH; do we really care about the internal variable naming convention? :)